test/parallel/test-http2-compat-method-connect.js

```diff
@@ -0,0 +1,41 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+const server = http2.createServer(common.mustNotCall());
+
+server.listen(0, common.mustCall(() => testMethodConnect(2)));
+
+server.once('connect', common.mustCall((req, res) => {
+  assert.strictEqual(req.headers[':method'], 'CONNECT');
+  res.statusCode = 405;
+  res.end();
+}));
+
+function testMethodConnect(testsToRun) {
+  if (!testsToRun) {
+    return server.close();
```

Comment:
Just pointing out... we now have a "countdown" utility that can be used to improve this kind of logic.

For instance,

```js
const Countdown = require('../common/countdown');
const countdown = new Countdown(2, () => console.log('done!') );
countdown.dec();
countdown.dec();
```

The callback based into Countdown will be called automatically and synchronously when the internal counter reaches zero.

Reply:
Thanks @jasnell, that's super useful. Will definitely use it in the future! Did you want me to update this test as well or good as is?

Reply:
It's good as is. We can convert over to the countdown util separately